python/pyiceberg/manifest.py

```diff
@@ -128,6 +128,10 @@ class ManifestFile(IcebergBaseModel):
     partitions: Optional[List[FieldSummary]] = Field()
     key_metadata: Optional[bytes] = Field()
 
+    def fetch_manifest_entry(self, io: FileIO) -> List[ManifestEntry]:
```

Comment:
Could this have a better name? I think this is more a read than a fetch.

I also think this is a bit too simple. These methods should not pull entire files of content into memory and should support projecting columns. Reading a manifest file can be really expensive if you load all of the file metadata. We should use a generator rather than a list.